File: LANG-go/infix_to_postfix.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"unicode"
)

var lookahead rune

func expr() {
	term()
	for true {
		if lookahead == '+' {
			match('+')
			term()
			fmt.Println("+")
		} else if lookahead == '-' {
			match('-')
			term()
			fmt.Println("-")
		} else {
			return
		}
	}
}

func term() {
	if unicode.IsNumber(lookahead) {
		t := lookahead
		match(lookahead)
		fmt.Println(string(t))
	} else {
		fmt.Printf("syntax error in term()! %v - is not number!\n", string(lookahead))
		os.Exit(1)
	}
}

func match(t rune) {
	if lookahead == t {
		fmt.Print("read:")
		lookahead = readOneRune(os.Stdin)
	} else {
		fmt.Println("syntax error in match()!")
		os.Exit(1)
	}
}

func readOneRune(rd io.Reader) rune {
	in := bufio.NewReader(rd)
	foo, _, _ := in.ReadRune()

	return foo
}

func main() {
	fmt.Print("read:")
	lookahead = readOneRune(os.Stdin)

	expr()
	fmt.Println()
}

File: LANG-python/infix_to_postfix.py

import numbers
from pickle import TRUE

lookahead: str = ""

def expr() -> None:
    term()
    while TRUE:
        if lookahead == "+":
            match("+")
            term()
            print("+")
        elif lookahead == "-":
            match("-")
            term()
            print("-")
        else:
            return

def term() -> None:
    if lookahead.isnumeric():
        t = lookahead
        match(lookahead)
        print(t)
    else:
        print("syntax error in term()! {} - is not number!".format(lookahead))
        exit()

def match(t: str) -> None:
    global lookahead
    if lookahead == t:
        print("read:", end="")
        lookahead = input()
    else:
        print("syntax error in match!")
        exit(0)


def main() -> None:
    global lookahead
    print("read:", end="")
    lookahead = input()

    expr()
    print()

if __name__ == "__main__":
    main()